src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/FaultDetectionMetricsSnapshot.java

```diff
@@ -0,0 +1,257 @@
+                                .and(DSL.field(Fields.ET.toString()).isNull())
+                                .and(DSL.field(Fields.ST.toString()).gt(this.windowStartTime - EXPIRE_AFTER)));
+    }
+
+    public SelectHavingStep<Record> groupByRidAndTypeSelect() {
+        ArrayList<SelectField<?>> fields =
+                new ArrayList<SelectField<?>>() {
+                    {
+                        this.add(DSL.field(DSL.name(Fields.RID.toString()), String.class));
+                        this.add(DSL.field(DSL.name(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString()), String.class));
+                        this.add(DSL.field(DSL.name(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString()), String.class));
+                        this.add(DSL.field(DSL.name(Fields.FAULT_DETECTION_TYPE.toString()), String.class));
+                    }
+                };
+        fields.add(
+                DSL.max(DSL.field(Fields.ST.toString(), Long.class)).as(DSL.name(Fields.ST.toString())));
+        fields.add(
+                DSL.max(DSL.field(Fields.ET.toString(), Long.class)).as(DSL.name(Fields.ET.toString())));
+        fields.add(
+                DSL.max(DSL.field(Fields.ERROR.toString(), Integer.class)).as(DSL.name(Fields.ERROR.toString())));
+        LOG.error("Initial - " + fetchAll().size());
+        LOG.error("Breakdown 1 - " + create
+                .select(fields)
+                .from(DSL.table(this.tableName))
+                .groupBy(DSL.field(Fields.RID.toString()), DSL.field(Fields.FAULT_DETECTION_TYPE.toString())).fetch().size());
+        return create
+                .select(fields)
+                .from(DSL.table(this.tableName))
+                .groupBy(DSL.field(Fields.RID.toString()), DSL.field(Fields.FAULT_DETECTION_TYPE.toString()));
+    }
+
+    public SelectHavingStep<Record> fetchLatencyTable() {
+        ArrayList<SelectField<?>> fields =
+                new ArrayList<SelectField<?>>() {
+                    {
+                        this.add(DSL.field(DSL.name(Fields.RID.toString()), String.class));
+                        this.add(DSL.field(DSL.name(AllMetrics.FaultDetectionDimension.SOURCE_NODE_ID.toString()), String.class));
+                        this.add(DSL.field(DSL.name(AllMetrics.FaultDetectionDimension.TARGET_NODE_ID.toString()), Long.class));
+                        this.add(DSL.field(DSL.name(Fields.FAULT_DETECTION_TYPE.toString()), String.class));
+                        this.add(DSL.field(Fields.ST.toString(), Long.class));
+                        this.add(DSL.field(Fields.ET.toString(), Long.class));
+                        this.add(DSL.field(Fields.ERROR.toString(), Integer.class));
+                    }
+                };
+        fields.add(
+                DSL.field(Fields.ET.toString())
+                        .minus(DSL.field(Fields.ST.toString()))
+                        .as(DSL.name(Fields.LAT.toString())));
+        LOG.error("Breakdown 2 - " + create
```

Comment:
Can you help me understand why this is an error log ?

Reply:
My bad. This was log used to debug an issue. Removed it